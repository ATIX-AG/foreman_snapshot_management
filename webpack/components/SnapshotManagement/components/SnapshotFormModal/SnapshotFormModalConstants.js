export const SNAPSHOT_FORM_MODAL = 'foremanSnapshotMgmtSnapshotFormModal';
