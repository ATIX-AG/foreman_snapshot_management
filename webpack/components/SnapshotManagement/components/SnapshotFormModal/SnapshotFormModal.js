import React from 'react';
import PropTypes from 'prop-types';

import ForemanModal from 'foremanReact/components/ForemanModal';
import { sprintf, translate as __ } from 'foremanReact/common/I18n';

import SnapshotForm from '../SnapshotForm';

import { SNAPSHOT_FORM_MODAL } from './SnapshotFormModalConstants';

const SnapshotFormModal = ({ host, setModalClosed, ...props }) => (
  <ForemanModal
    id={SNAPSHOT_FORM_MODAL}
    title={sprintf(__('Create Snapshot for %s'), host.name)}
    enforceFocus
  >
    <ForemanModal.Header closeButton={false} />

    <div>
      <SnapshotForm
        setModalClosed={setModalClosed}
        hostId={host.id}
        {...props}
      />
    </div>
  </ForemanModal>
);

SnapshotFormModal.propTypes = {
  host: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  setModalClosed: PropTypes.func.isRequired,
};

export default SnapshotFormModal;
