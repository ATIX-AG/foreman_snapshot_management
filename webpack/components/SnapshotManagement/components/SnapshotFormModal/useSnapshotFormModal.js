import { useForemanModal } from 'foremanReact/components/ForemanModal/ForemanModalHooks';

import { SNAPSHOT_FORM_MODAL } from './SnapshotFormModalConstants';

const useSnapshotFormModal = () => useForemanModal({ id: SNAPSHOT_FORM_MODAL });

export default useSnapshotFormModal;
