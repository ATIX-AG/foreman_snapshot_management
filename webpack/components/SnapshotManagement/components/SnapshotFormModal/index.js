import React from 'react';

import useSnapshotFormModal from './useSnapshotFormModal';
import SnapshotFormModal from './SnapshotFormModal';

const WrappedSnapshotFormModal = props => {
  const { setModalClosed } = useSnapshotFormModal();

  return <SnapshotFormModal setModalClosed={setModalClosed} {...props} />;
};

export default WrappedSnapshotFormModal;
