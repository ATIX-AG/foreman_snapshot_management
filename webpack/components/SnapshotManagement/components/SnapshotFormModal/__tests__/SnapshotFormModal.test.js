import {
  testComponentSnapshotsWithFixtures,
  // shallowRenderComponentWithFixtures,
} from 'react-redux-test-utils';
// import configureStore from 'redux-mock-store';

import SnapshotFormModal from '../SnapshotFormModal';

// const mockStore = configureStore([]);

const setModalClosed = () => null;
const fixtures = {
  normal: { host: { id: 42, name: 'deep.thought' }, setModalClosed },
};

describe('SnapshotFormModal', () => {
  describe('renders', () =>
    testComponentSnapshotsWithFixtures(SnapshotFormModal, fixtures));
});
