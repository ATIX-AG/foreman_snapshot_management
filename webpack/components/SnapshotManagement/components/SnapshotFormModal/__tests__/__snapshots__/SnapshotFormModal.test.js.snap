// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotFormModal renders normal 1`] = `
<ConnectedForemanModal
  enforceFocus={true}
  id="foremanSnapshotMgmtSnapshotFormModal"
  title="Create Snapshot for deep.thought"
>
  <ForemanModalHeader
    closeButton={false}
  />
  <div>
    <WrappedSnapshotForm
      hostId={42}
      setModalClosed={[Function]}
    />
  </div>
</ConnectedForemanModal>
`;
